### PR TITLE
Explain behavior of `config.get`

### LINKED ISSUE #3969: Explain behavior of `config.get`
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

Docs to update:
- https://docs.getdbt.com/reference/dbt-jinja-functions/config#configget
- https://docs.getdbt.com/reference/resource-configs/where#custom-logic

https://github.com/dbt-labs/docs.getdbt.com/pull/3689 was merged, which might already address the 2nd bullet point.

### What part(s) of the page would you like to see updated?

See here for background and details:
- https://github.com/dbt-labs/dbt-core/issues/8047#issuecomment-1624460809

### Additional information

In an [internal slack thread](https://dbt-labs.slack.com/archives/C044WU55X7Y/p1692797280367669?thread_ts=1692797208.243079&cid=C044WU55X7Y), @gshank noted that:
> Instead of `config.get("some_name", "some_default")` it works with built-in configs to do `config.get("some_name") or "some_default"` instead

So if I'm interpreting that correctly, then we'd recommend doing this to get behavior that is similar to [`coalesce()`](https://docs.snowflake.com/en/sql-reference/functions/coalesce):
```sql
{% set where = config.get("where") or "your_desired_default_goes_here" %}
```

### History

Here's some historical issues related to how we've reach the current implementation:
- https://github.com/dbt-labs/dbt-core/issues/8047
- https://github.com/dbt-labs/dbt-core/issues/6976
- https://github.com/dbt-labs/dbt-core/issues/6309
- https://github.com/dbt-labs/dbt-core/issues/4273
- https://github.com/dbt-labs/dbt-core/issues/2101

### PR DESCRIPTION
**Pull request by dbeatty10**
resolves https://github.com/dbt-labs/docs.getdbt.com/issues/3969

## What are you changing in this pull request and why?

Details in https://github.com/dbt-labs/docs.getdbt.com/issues/3969

#### TLDR

The key insight is that many people expect `config.get("foo", default="bar")` to act like [`coalesce`](https://docs.snowflake.com/en/sql-reference/functions/coalesce) in SQL, but it does **not**.

There are cases where users will want to use an alternative value whenever `config.get("something")` is `None` (similar to `coalesce` in SQL). This PR documents how to do that.

## Checklist
- [x] I have reviewed the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.

---
🚀 Deployment available! Here are the direct links to the updated files:

- https://docs-getdbt-com-git-dbeatty10-patch-2-dbt-labs.vercel.app/reference/dbt-jinja-functions/config

### CODE REVIEW

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 1]
```diff
@@ -34,13 +34,21 @@ __Args__:
 
 The `config.get` function is used to get configurations for a model from the end-user. Configs defined in this way are optional, and a default value can be provided.
 
+There are 3 cases:
+1. The configuration variable exists, it is it not `None`
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md]:
```suggestion
1. The configuration variable exists, it is not `None`
```

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 2]
```diff
@@ -34,13 +34,21 @@ __Args__:
...
+1. The configuration variable exists, it it is `None`
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md]:
```suggestion
1. The configuration variable exists, it is `None`
```

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 3]
```diff
@@ -34,13 +34,21 @@ __Args__:
 
 The `config.get` function is used to get configurations for a model from the end-user. Configs defined in this way are optional, and a default value can be provided.
 
+There are 3 cases:
+1. The configuration variable exists, it is not `None`
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md:38]:
```suggestion
1. The configuration variable exists, it isn't `None`
```

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 4]
```diff
@@ -34,13 +34,21 @@ __Args__:
 
 The `config.get` function is used to get configurations for a model from the end-user. Configs defined in this way are optional, and a default value can be provided.
 
+There are 3 cases:
+1. The configuration variable exists, it isn't `None`
+1. The configuration variable exists, it is `None`
+1. The configuration variable does not exist
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md:40]:
```suggestion
1. The configuration variable doesn't exist
```

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 5]
```diff
@@ -34,13 +34,21 @@ __Args__:
 
 The `config.get` function is used to get configurations for a model from the end-user. Configs defined in this way are optional, and a default value can be provided.
 
+There are 3 cases:
+1. The configuration variable exists, it isn't `None`
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md]:
```suggestion
1. The configuration variable exists, it is not `None`
```

File: website/docs/reference/dbt-jinja-functions/config.md [hunk 6]
```diff
@@ -34,13 +34,21 @@ __Args__:
 
 The `config.get` function is used to get configurations for a model from the end-user. Configs defined in this way are optional, and a default value can be provided.
 
+There are 3 cases:
+1. The configuration variable exists, it is not `None`
+1. The configuration variable exists, it is `None`
+1. The configuration variable doesn't exist
```
Comment (mirnawong1) [inline website/docs/reference/dbt-jinja-functions/config.md]:
```suggestion
1. The configuration variable does not exist
```

**Review by mirnawong1 — APPROVED** [review]
made some small typos and looks good! thanks @dbeatty10 !

**Comment by dbeatty10** [comment]
> made some small typos and looks good! thanks @dbeatty10 !

@mirnawong1 thank you so much for fixing this up 🧠 !

For such a small PR, I sure had a high density of typos 😅 -- thanks for having my back and getting it on track! 🛤️
